app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/TimeUtils.scala

```diff
@@ -99,23 +100,35 @@ object TimeUtils {
           day ← toValidInt(dayKey, dateFieldStrings._1, Some(31))
           month ← toValidInt(monthKey, dateFieldStrings._2, Some(12))
           year ← toValidInt(yearKey, dateFieldStrings._3, None)
-          date ← Either
-                   .fromTry(Try(LocalDate.of(year, month, day)))
-                   .leftMap(_ => FormError(dateKey, "error.invalid"))
-                   .flatMap(date =>
-                     if (maximumDateInclusive.exists(_.isBefore(date)))
-                       Left(FormError(dateKey, "error.tooFarInFuture"))
-                     else if (minimumDateInclusive.exists(_.isAfter(date)))
-                       Left(FormError(dateKey, "error.tooFarInPast"))
-                     else if (date.isBefore(minimumDate))
-                       Left(FormError(dateKey, "error.before1900"))
-                     else
-                       extraValidation
-                         .map(_(date))
-                         .find(_.isLeft)
-                         .getOrElse(Right(()))
-                         .map(_ => date)
-                   )
+          date ←
+            Either
+              .fromTry(Try(LocalDate.of(year, month, day)))
+              .leftMap(_ => FormError(dateKey, "error.invalid"))
+              .flatMap(date =>
+                if (maximumDateInclusive.exists(_.isBefore(date)))
+                  Left(FormError(dateKey, "error.tooFarInFuture"))
+                else if (
+                  (dateKey === "completionDate") && minimumDateInclusive
+                    .exists(_.isAfter(date))
+                )
+                  Left(
+                    FormError(
+                      dateKey,
+                      "error.tooFarInPastWithArg",
+                      List(govDisplayFormat(minimumDateInclusive.getOrElse(LocalDate.now())))
+                    )
+                  )
```

Comment:
We can use existing condition and pass parameter. I think its good to use generic validation, instead of field specific validation.